warehouse/macaroons/services.py

```diff
@@ -120,27 +120,42 @@ def verify(self, raw_macaroon, context, principals, permission):
         if dm is None:
             raise InvalidMacaroonError("deleted or nonexistent macaroon")
 
-        verifier = Verifier(m, context, principals, permission)
+        verifier = Verifier(m, context, principals, permission, identity)
```

Comment:
Flagging for review: I plumbed `request.identity` through here, since the macaroon AuthZ layer needs to know the current identity to pass/fail the token. There might be a cleaner way to do this, although I suspect that we can clean this up more generally by removing the legacy `IAuthorizationPolicy` implementations and doing everything directly though `ISecurityPolicy`.

Reply:
Stale.